.github/workflows/deploy_infra.yml

```diff
@@ -0,0 +1,146 @@
+name: Deploy to Infrastructure
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: Environment to deploy (dev, stage, prod)
```

Comment:
I think we may want to expand the environment description to indicate it is used for github action environment and aws account type